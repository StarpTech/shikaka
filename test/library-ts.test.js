const { prepareLibraryTest } = require('./utils');

describe('library-ts fixture', () => {
  test('build with shikaka', async () => {
    const {
      files: { Button, styles }
    } = await prepareLibraryTest('library-ts', 'src/index.ts');

    expect(Button).toMatch(`import React from 'react';

var styles = {"button":"Button__button"};

// eslint-disable-next-line no-unused-vars
function Button(props) {
  return /*#__PURE__*/React.createElement("button", {
    className: styles.button
  }, props.children);
}

export default Button;
`);
    expect(styles).toMatch(`.Button__button {
  display: grid;
  transition: all .5s;
  user-select: none;
  background: linear-gradient(to bottom, white, black);
}`);
  });
});
