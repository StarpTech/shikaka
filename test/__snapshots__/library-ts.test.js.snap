// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`library-ts fixture build with shikaka 1`] = `
"Used script: node cli.js src/index.ts --out-dir test/fixtures/library-ts/dist --root-dir test/fixtures/library-ts

Directory tree:

library-ts
  dist
    Button.js
    index.js
    styles.css
  package.json
  src
    components
      Button
        index.module.css
        index.tsx
    index.ts
"
`;

exports[`library-ts fixture build with shikaka 2`] = `3`;
